web/views.py

```
@@ -6961,6 +6983,57 @@ def delete_post(request, post_id):
     return redirect("social_media_dashboard")
 
 
+def generate_discount_code(length=8):
+    return "".join(random.choices(string.ascii_uppercase + string.digits, k=length))
+
+
+@login_required
+def apply_discount_via_referrer(request) -> HttpResponse:
```

Ruff (0.8.2)

Missing type annotation for function argument `request`

(ANN001)